### PR TITLE
Vitepress conversion of docs.

### PR DESCRIPTION
**Pull request by tzarc**
## Description

Each commit is a group of related changes -- all links being fixed, all info/warning panels, etc. etc.

`qmk docs` and `qmk generate-docs` should function, so long as `yarn` is present in the environment.
Now requires adding a symlink in order to do static docs generation (`qmk generate-docs`), so Windows may choke. `qmk docs` should work fine on Windows, though.

Opening as draft while I automate the rest of the conversions as per `__capabilities.md`.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [x] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [x] Documentation

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
Closing while I fight GHA. Seems not to execute while an open PR is present.

**Comment by tzarc** [comment]
Should be good to go at this point.

**Review by drashna — APPROVED** [review]

File: .github/workflows/docs.yml [hunk 1]
```diff
@@ -29,18 +35,51 @@ jobs:
...
+        nvm use 20
+        corepack enable
         npm install -g moxygen
 
     - name: Build docs
       run: |
+        source $HOME/.bashrc
+        nvm use 20
         qmk --verbose generate-docs
+        touch '.build/docs/.nojekyll'
```
Comment (KarlK90) [inline .github/workflows/docs.yml:56]: Does this still apply or can it be removed in the final merge?

File: .github/workflows/docs.yml [hunk 2]
```diff
@@ -29,18 +35,51 @@ jobs:
...
+      if: github.repository == 'tzarc/qmk_firmware'
       uses: JamesIves/github-pages-deploy-action@v4.6.1
       with:
-        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
-        BASE_BRANCH: master
-        BRANCH: gh-pages
-        FOLDER: .build/docs
-        GIT_CONFIG_EMAIL: hello@qmk.fm
+        branch: gh-pages
+        folder: .build/docs
```
Comment (KarlK90) [inline .github/workflows/docs.yml:85]: Does this still apply or can it be removed in the final merge?

File: .github/workflows/docs.yml [hunk 3]
```diff
@@ -29,18 +35,51 @@ jobs:
...
+      if: github.repository == 'qmk/qmk_firmware'
+      run: |
+        # Override target CNAME
+        echo 'docs.qmk.fm' > .build/docs/CNAME
+
+    - name: Override CNAME
+      if: github.repository == 'tzarc/qmk_firmware'
+      run: |
+        # Temporarily override target CNAME during development
+        echo 'vitepress.qmk.fm' > .build/docs/CNAME
```
Comment (KarlK90) [inline .github/workflows/docs.yml:68]: Does this still apply or can it be removed in the final merge?

File: .github/workflows/docs.yml [hunk 4]
```diff
@@ -7,20 +7,26 @@ on:
...
+    shell: bash
+
 jobs:
   generate:
     runs-on: ubuntu-latest
     container: ghcr.io/qmk/qmk_cli
 
     # protect against those who develop with their fork on master
-    if: github.repository == 'qmk/qmk_firmware'
+    if: github.repository == 'qmk/qmk_firmware' || (github.repository == 'tzarc/qmk_firmware' && github.ref == 'refs/heads/vitepress')
```
Comment (KarlK90) [inline .github/workflows/docs.yml:29]: Does this still apply or can it be removed in the final merge?

**Review by KarlK90 — CHANGES_REQUESTED** [review]

File: .github/workflows/docs.yml (hunk 3 above)
Comment (tzarc) [reply .github/workflows/docs.yml:68]: Will be done in a follow up PR. Intentionally written such that it works on both repos.

**Pull request merged**